src/main/java/org/wildfly/client/config/ConfigurationXMLStreamReader.java

```diff
@@ -725,13 +727,61 @@ default Expression getExpressionAttributeValue(int index, Expression.Flag... fla
         final String attributeValue = getAttributeValue(index);
         if (attributeValue == null) {
             return null;
+        } else if (attributeValue.startsWith("env.")) {
+            String envVar = System.getenv().get(attributeValue.substring(4));
+
+            if (envVar.contains("ENC:")) {
+                return resolveEncryptedExpression(envVar, flags);
+            } else {
+                return Expression.compile(envVar, flags);
+            }
```

Comment:
This same if-block 733-737 is used in 4 places in this method.
I suggest moving it into a private method.

Reply:
No need.  It would be a nice to have when someone else adds to the file in the future.